### PR TITLE
Implement attributesToSearchOn

### LINKED ISSUE #457: [v1.3] Define fields to search on at search-time (`attributesToSearchOn`)
⚠️ This issue is generated, it means the examples and the namings do not necessarily correspond to the language of this repository.
Also, if you are a maintainer, please add any clarification and instructions about this issue.

Sorry if this is already wholly/partially implemented. Feel free to let me know about the state of this issue in the repo.

Related to https://github.com/meilisearch/integration-guides/issues/280

----

This issue is divided into two sections, first, you need to make the implementation, and second, you must update the code-samples (no one likes outdated docs, right?).

### New implementation

Related to:
- issue: https://github.com/meilisearch/meilisearch/issues/3772
- discussion: https://github.com/meilisearch/product/discussions/107#discussioncomment-6184044
- spec: https://github.com/meilisearch/specifications/pull/251

Add support for receiving a new key, `attributesToSearchOn ` (an array of field names), to enable running searches over a subset of `searchableAttributes` without modifying Meilisearch’s index settings.

### Code samples

Inside of this file: `.code-samples.meilisearch.yml`:

1. Create a new entry with this key `search_parameter_guide_attributes_to_search_on_1` containing a call to the `search` method using the new behavior from the index `movies` using `attributesToSearchOn` with an array of one string `["overview"]`.
Add a new code sample to the `.code-samples.meilisearch.yml` file:

Use this as a reference _if the previous description was not helpful_:

```
search_parameter_guide_attributes_to_search_on_1: |-
  POST 'http://localhost:7700/indexes/movies/search' 
  with data: { 
    "q": "adventure", "attributesToSearchOn": ["overview"] 
  }   
```

TODO:
- [ ] Add the ability to receive a new param in the `search` method called `attributesToSearchOn`.
- [ ] Add integration tests
- [ ] Update the code-samples accordingly

### PR DESCRIPTION
**Pull request by SKeeneCode**
# Pull Request

## Related issue
Fixes #457

## What does this PR do?
- Implements AttributesToSearchOn in SearchQuery

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -329,6 +329,12 @@ search_parameter_guide_show_matches_position_1: |-
     {
         ShowMatchesPosition = True,
     });
+search_parameter_guide_attributes_to_search_on_1: |-
+  var searchQuery = new SearchQuery
+  {
+    AttributesToSearchOn = new[] { "overview" }
+  };
+  await client.Index("movies").SearchAsync<Movie>("exciting", searchQuery);
```
Comment (brunoocasali) [inline .code-samples.meilisearch.yaml]:
```suggestion
  await client.Index("movies").SearchAsync<Movie>("adventure", searchQuery);
```

**Review by brunoocasali — CHANGES_REQUESTED** [review]
Just a typo in the code-samples and we are ready to go!

**Comment by SKeeneCode** [comment]
Tests work locally but failed in the workflow with the following error:

```
Failed Meilisearch.Tests.ServerConfigs.BaseUriServer+SearchTests.CustomSearchWithAttributesToSearchOnNoResults [2 ms]
  Error Message:
   Meilisearch.MeilisearchApiError : MeilisearchApiError, Message: Unknown field `attributesToSearchOn`: expected one of `q`, `offset`, `limit`, `page`, `hitsPerPage`, `attributesToRetrieve`, `attributesToCrop`, `cropLength`, `attributesToHighlight`, `showMatchesPosition`, `filter`, `sort`, `facets`, `highlightPreTag`, `highlightPostTag`, `cropMarker`, `matchingStrategy`,
```

Locally with an instance set up with just:

```
curl -L https://install.meilisearch.com | sh
./meilisearch
```

Tests pass in my IDE: 

image

I notice the field `attributesToSearchOn` isn't included in the expect list in the error. Is the workflow spooling up an out of date version @brunoocasali ?

**Comment by ahmednfwela** [comment]
@SKeeneCode you need to change this line to 1.3.0
https://github.com/meilisearch/meilisearch-dotnet/blob/main/.env#L1

@brunoocasali can we change this to `latest` to grab the latest available meilisearch version ?

**Comment by SKeeneCode** [comment]
@ahmednfwela thanks for pointing that out ^.^ I've changed it to 1.3.0 but if brunoo says latest is fine I can do change it to that.

**Comment by brunoocasali** [comment]
Since v0.30 we have updated this version number, but I don't remember why. Let's change back to `latest` and if this is not what we expect, then we can go back to `v1.3.0` :D

**Review by brunoocasali — COMMENTED** [review]
Thanks a lot for your contribution!

bors merge

**Review by brunoocasali — APPROVED** [review]

**Pull request merged**